app/initializers/fastboot.js

```diff
@@ -4,62 +4,18 @@ export default {
   name: "fast-boot",
 
   initialize: function(App) {
-    // Detect if we're running in Node. If not, there's nothing to do.
-    if (typeof document === 'undefined') {
-      var doc = new SimpleDOM.Document();
-      var domHelper = new Ember.HTMLBars.DOMHelper(doc);
-
-      domHelper.protocolForURL = function(url) {
+    // Detect if we're running in FastBoot mode. If not, there's nothing to do.
+    if (FastBoot) {
```

Comment:
@stefanpenner Good point (currently this initializer is run in both fastboot and non-fastboot modes), on certain browsers (looking at you IE) attempting to access an unprefixed global (without using `typeof`) results in an error...

As an aside, we could easily make the `treeForApp` hook a noop if not running the `ember fastboot` command (which would mean apps do not pay any filesize penalty when using `ember build` or `ember s`).